src/engine/renderer/glsl_source/forwardLighting_fp.glsl

```diff
@@ -975,12 +975,13 @@ void	main()
 	}
 	diffuse.rgb *= u_LightColor * NL;
 
+#if !defined(USE_PHYSICAL_SHADING)
```

Comment:
Is this supposed to say r_physicalshading?

Also you could combine these into a single #if.

Reply:
No, both specular and physical textures can be rendered in the same screen. This must generate two GLSL shaders, one for specular path, one for material path, those two paths must be available together.

In fact the real logic is:

```c
#ifdef USE_PHYSICAL_SHADING
#ifdef r_physicalMapping
// some code
#endif // r_physicalMapping
#else !USE_PHYSICAL_SHADING
#ifdef r_specularMapping
// some code
#endif // r_specularMapping
#endif !USE_PHYSICAL_SHADING
```

And I already combined the two ifs for `USE_PHYSICAL_SHADING` and `r_physicalMapping`.

Reply:
In this glsl code the physical shading is just not implemented (it's the forward renderer we don't use), that's why the `#if !defined(USE_PHYSICAL_SHADING)`: this is just to be sure a possible PBR map is not used by specular computation, but we still have to test if it's PBR map (to never do anything) or a specular map (to do something only if `r_specularMapping` is enabled).

Reply:
Well, I discovered we cannot combine `USE_PHYSICAL_SHADING` and `r_physicalMapping` otherwise the PBR map would be used as specular map when `r_physicalMapping` is disabled.

Reply:
fixed